kasa/smartprotocol.py

```diff
@@ -64,6 +75,22 @@ async def query(self, request: Union[str, Dict], retry_count: int = 3) -> Dict:
         """Query the device retrying for retry_count on failure."""
         async with self._query_lock:
             resp_dict = await self._query(request, retry_count)
+
+            if (
+                error_code := ErrorCode(resp_dict.get("error_code"))
+            ) != ErrorCode.SUCCESS:
```

Comment:
Just a personal opinion, but this would read much cleaner to me:
```
if error_code < 0:
    error_code = ErrorCode(resp_dict.get("error_code"))
    ...
```

Reply:
I'd have to get the `error_code_int` first from the dict and then create the ErrorCode so I don't think it would read as neatly as the above snippet.  Making mypy happy will probably make it look worse as well.  Hopefully we're ok to skip this one.